lib/matplotlib/axes/_base.py

```diff
@@ -29,6 +30,54 @@
 _log = logging.getLogger(__name__)
 
 
+def _axis_method_wrapper(attr_name, method_name, *, doc_sub=None):
+    """
+    Helper to generate Axes methods wrapping Axis methods.
+
+    After ::
+
+        get_foo = _axis_method_wrapper("xaxis", "get_bar")
+
+    ``get_foo`` is a method that forwards it arguments to the ``get_bar``
+    method of the ``xaxis`` attribute, and gets its signature and docstring
+    from ``Axis.get_bar``.
+
+    The docstring of ``get_foo`` is built by replacing "this Axis" by "the
+    {attr_name}" ("the xaxis", "the yaxis") in the wrapped method's docstring;
+    additional replacements can by given in *doc_sub*.  The docstring is also
+    dedented to simplify further manipulations.
+    """
+
+    get_method = attrgetter(f"{attr_name}.{method_name}")
+
+    def wrapper(self, *args, **kwargs):
+        return get_method(self)(*args, **kwargs)
+
+    # Manually copy the signature instead of using functools.wraps because
+    # displaying the Axis method source when asking for the Axes method source
+    # would be confusing.
+    wrapped_method = getattr(maxis.Axis, method_name)
+    wrapper.__signature__ = inspect.signature(wrapped_method)
+    doc = wrapped_method.__doc__
+    if doc:
+        doc_sub = {"this Axis": f"the {attr_name}", **(doc_sub or {})}
+        for k, v in doc_sub.items():
+            assert k in doc, \
+                (f"The docstring of wrapped Axis method {method_name!r} must "
+                 f"contain {k!r} as a substring.")
+            doc = doc.replace(k, v)
+        wrapper.__doc__ = inspect.cleandoc(doc)
+
+    class Placeholder:
+        def __set_name__(self, owner, name):
+            wrapper.__module__ = owner.__module__
+            wrapper.__name__ = name
+            wrapper.__qualname__ = f"{owner.__qualname__}.{name}"
+            setattr(owner, name, wrapper)
```

Comment:
Does this place holder replace it's self on the wraper on the class as part of class definition?   I assume this is done to make sure the reported names are all correct?

Reply:
Yes.